src/vos/vos_dtx_iter.c

```diff
@@ -52,8 +52,6 @@ dtx_iter_fini(struct vos_iterator *iter)
 	struct vos_dtx_iter	*oiter = iter2oiter(iter);
 	int			 rc = 0;
 
-	D_ASSERT(iter->it_type == VOS_ITER_DTX);
```

Comment:
How can this happen? It looks a serious bug to me.

Reply:
dtx_iter_prep() => dbtree_iter_prepare(), return failure, then dtx_iter_prep () will call dtx_iter_fini().

Reply:
So similar issue exists in all other types (CONT, OBJ), shouldn't we fix them all together?